k4FWCore/scripts/k4run

```diff
@@ -104,12 +104,20 @@ if __name__ == "__main__":
                         propnargs = "+"
                         
                       propName = conf.name() + '.' + prop
-                      parser.add_argument("--%s" % propName, type=proptype, help=props[prop][1],
+                      # try to add argument once as a shortcut (prop, p.ex '--filename')
+                      # and once with the full property name (propName, p.ex '--PodioOutput.filename')
+                      # if the argument is already present, an ArgumentError is raised that is dealt with below
+                      parser.add_argument("--%s" % prop, "--%s" % propName, type=proptype, help=props[prop][1],
```

Comment:
Because of the "completion" argparse does by itself (https://docs.python.org/3/library/argparse.html#allow-abbrev). what about adding 
`nameProp = prop + "." + conf.name()`
Then you can still use --filename if there is no other property with that name and argparse will complete to `--filename.PodioOutput`, but otherwise you get an error, which is probably preferable to setting the wrong option?

Might be better to put the propName first in the list of arguments to have a better visual grouping of the options when doing `--help`

Reply:
I like that a lot! It feels slightly backwards to begin with prop, but it will simplify things a lot and that's always good.

Reply:
This won't actually work with `--filename` because of the ambiguity with `--filenameRemote` though. I think we can just rename the properties to reduce the ambiguities, however

Reply:
I meant "adding" as in keeping the "conf.Name().prop" still available. And using that as the "primary" argument for the "visual" grouping of options

```python
confProp = conf.Name() + "." + prop
propConf = prop + "." + conf.Name()
parser.add_argument("--%s" % confProp,
    "--%s" % propConf,
    type=proptype, help=props[prop][1])
```

e.g.:
```
  --InitDD4hep.Blocking [INITDD4HEP.BLOCKING], --Blocking.InitDD4hep [INITDD4HEP.BLOCKING]
                        if algorithm invokes CPU-blocking system calls
                        (offloads computations to accelerators or quantum
                        processors, performs disk or network I/O, is bound by
                        resource synchronization, etc) [Gaudi::Algorithm]
  --InitDD4hep.PrintEmptyCounters [INITDD4HEP.PRINTEMPTYCOUNTERS], --PrintEmptyCounters.InitDD4hep [INITDD4HEP.PRINTEMPTYCOUNTERS]
                        force printing of empty counters, otherwise only
                        printed in DEBUG mode [GaudiCommon<Algorithm>]
  --InitDD4hep.ProcessorType [INITDD4HEP.PROCESSORTYPE], --ProcessorType.InitDD4hep [INITDD4HEP.PROCESSORTYPE]
                        [MarlinProcessorWrapper]
  --InitDD4hep.Parameters INITDD4HEP.PARAMETERS [INITDD4HEP.PARAMETERS ...], --Parameters.InitDD4hep INITDD4HEP.PARAMETERS [INITDD4HEP.PARAMETERS ...]
                        [MarlinProcessorWrapper]
  --VXDBarrelDigitiser.Blocking [VXDBARRELDIGITISER.BLOCKING], --Blocking.VXDBarrelDigitiser [VXDBARRELDIGITISER.BLOCKING]
                        if algorithm invokes CPU-blocking system calls
                        (offloads computations to accelerators or quantum
                        processors, performs disk or network I/O, is bound by
                        resource synchronization, etc) [Gaudi::Algorithm]
  --VXDBarrelDigitiser.PrintEmptyCounters [VXDBARRELDIGITISER.PRINTEMPTYCOUNTERS], --PrintEmptyCounters.VXDBarrelDigitiser [VXDBARRELDIGITISER.PRINTEMPTYCOUNTERS]
                        force printing of empty counters, otherwise only
                        printed in DEBUG mode [GaudiCommon<Algorithm>]
  --VXDBarrelDigitiser.ProcessorType [VXDBARRELDIGITISER.PROCESSORTYPE], --ProcessorType.VXDBarrelDigitiser [VXDBARRELDIGITISER.PROCESSORTYPE]
                        [MarlinProcessorWrapper]
  --VXDBarrelDigitiser.Parameters VXDBARRELDIGITISER.PARAMETERS [VXDBARRELDIGITISER.PARAMETERS ...], --Parameters.VXDBarrelDigitiser VXDBARRELDIGITISER.PARAMETERS [VXDBARRELDIGITISER.PARAMETERS ...]
                        [MarlinProcessorWrapper]
```